### PR TITLE
[Sofa.Testing][SofaGeneralEngine] Dispatch engine tests

### PR DESCRIPTION
**Pull request by fredroy**
Based on
 - #2650 

Generic tests on engine is messy, this PR
 - add template test class in Sofa.Testing.Tool
 - put back generic tests on engine in Sofa.Core (and remove deps on scenecreator)
 - clean sofageneralengine generic tests file
 - clean deps here and there

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by damienmarchal** [comment]
Hi Fred, 

The splitting seems to perpetuate the spurious dependency from SceneCreator to sofa.testing.tools. 
Maybe it is the time to remove the dependency to SceneCreator and use simpleapi in place so that sofa.testing is cleaned from SceneCreator.

**Comment by fredroy** [comment]
> Hi Fred,
> 
> The splitting seems to perpetuate the spurious dependency from SceneCreator to sofa.testing.tools. Maybe it is the time to remove the dependency to SceneCreator and use simpleapi in place so that sofa.testing is cleaned from SceneCreator.

I have to admit that I did not see that the DataEngineTestCreation.h file was using scenecreator itself, and this is a good idea to do it while sofang'ing. But I would prefer to do that in a further PR to not block my other sofang PRs 😗.
Except if it is trivial to remove/replace 🙃

File: modules/SofaGeneralEngine/SofaGeneralEngine_test/CMakeLists.txt [hunk 1]
```diff
@@ -26,16 +23,10 @@ list(APPEND SOURCE_FILES
     IndexValueMapper_test.cpp
     JoinPoints_test.cpp
     RandomPointDistributionInSurface_test.cpp
-    Engine_test.cpp
-    TestEngine.cpp
+    CommonSofaGeneralEngine_test.cpp
```
Comment (guparan) [inline modules/SofaGeneralEngine/SofaGeneralEngine_test/CMakeLists.txt:26]: Heavy renaming :skull: 
Why not `DataEngine_test.cpp` ?

File: modules/SofaGeneralEngine/SofaGeneralEngine_test/CMakeLists.txt [hunk 2]
```diff
@@ -2,13 +2,10 @@ cmake_minimum_required(VERSION 3.12)
...
 project(SofaGeneralEngine_test)
 
-set(HEADER_FILES
-    TestEngine.h
-    DataEngineTestCreation.h
-)
-
 set(SOURCE_FILES ../../empty.cpp)
 
+set(HEADER_FILES DataEngineTestCreation.h)
```
Comment (guparan) [inline modules/SofaGeneralEngine/SofaGeneralEngine_test/CMakeLists.txt:7]: Could be renamed `DataEngine_test.h` following my other comment.

File: modules/SofaGeneralEngine/SofaGeneralEngine_test/CMakeLists.txt (hunk 1 above)
Comment (damienmarchal) [reply modules/SofaGeneralEngine/SofaGeneralEngine_test/CMakeLists.txt:26]: CommonEngine_test :)

**Comment by fredroy** [comment]
This PR is deprecated due to 
- #2873

**Pull request closed without merging**